### PR TITLE
Added DebugServerInterceptor

### PR DESCRIPTION
**Pull request by SaiRepos**
- Added `DebugServerInterceptor`
- Renamed `DebugClientIntercetorTest` to `DebugInterceptorTest` for accommodating tests for both Client and Server debug interceptors.

### CODE REVIEW

**Comment by SaiRepos** [comment]
Hi @rmichela . Can you please review the PR and let me know for any changes required. Thanks!

File: contrib/grpc-contrib/src/main/java/com/salesforce/grpc/contrib/interceptor/DebugServerInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+            @Override
+            public void onMessage(ReqT message) {
+                logMessage(REQUEST, message);
+                super.onMessage(message);
+            }
+
+            @Override
+            public void onCancel() {
+                log(String.format("Call for method %s cancelled", call.getMethodDescriptor().getFullMethodName()));
```
Comment (rmichela) [inline contrib/grpc-contrib/src/main/java/com/salesforce/grpc/contrib/interceptor/DebugServerInterceptor.java:71]: Cancel should get its own level

Comment (SaiRepos) [reply contrib/grpc-contrib/src/main/java/com/salesforce/grpc/contrib/interceptor/DebugServerInterceptor.java]: Added the Level `Cancel` to log cancellation

**Comment by SaiRepos** [comment]
Hi @rmichela Could you please review the updates I committed to add `Cancel` Level.

**Pull request merged**